packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -233,7 +253,11 @@ export function createOdspNetworkError(
                 errorMessage, { canRetry: true, retryAfterMs }, driverProps);
             break;
     }
-    enrichOdspError(error, response, facetCodes, undefined, redirectLocation);
+
+    if (redirectLocation !== undefined) {
```

Comment:
It feels a bit weird (and complex) that we convert any error first to OdspRedirectError, and then to LocationRedirectionError. Can we avoid it? Not sure if we need to change a process, but if we can't avoid otherwise, I'd rather move conversion code EpochTracker.fetchCore() to here and get rid of OdspRedirectError

Reply:
we thought about just using redirectLocationError but the problem is we can't get the redirectUrl over here, that's only in epochTracker

Reply:
LocationRedirectionError is not Odsp specific. This property is only extracted in case of 404 error. So we can add this  in omitProperty arg in the 404 error class which is "NetworkErrorBasic"

Reply:
This logic should go up to the switch statement in previous function where we first find the redirectLocation on the error response.

Reply:
> we can add this in omitProperty arg in the 404 error class which is "NetworkErrorBasic"

We discussed this but opted for a separate error class, because that would leak ODSP-specific implementation info into driver base code.

Reply:
But redirectionLocation is not odsp specific. It is for all drivers although currently it is emitted by odsp only. But it is general.

Reply:
> But redirectionLocation is not odsp specific

Gotcha.  Then yeah, let's add this "redirectLocation" to omit set on `NetworkErrorBasic` or whatever - like @chensixx suggested a long time ago :P

Bonus points if we can move the declaration of this property to an interface in base driver instead of ODSP driver definitions, as close to the code that omits it from logging as possible (maybe even use `keyof` to tie it together at compile time)

Reply:
Ok I took a closer look at the code and discussed again with Jatin - sorry to go back and forth, but I stand by my original point that `redirectLocation` is an ODSP implementation detail.

Here's why: In the ODSP driver, we have multiple packages. odsp-doclib-utils sits as the interface with the service, and knows how to parse the error response to get this info, but it doesn't know how to turn that into a proper IOdspResolvedUrl.  That logic resides in the odsp-driver package.  So we need some intermediate type to pass the redirect location pulled off the HTTP response to the code in epochTracker (in odsp-driver).  That's this new class.  We can rename it to `OdspRedirectResponseError` or something that makes it more explicit that it's modeling the response we got.  Then it gets transformed into the `ILocationRedirectError` that we expose to hosts.

Consider if FRS has this same scenario one day.  Maybe all their driver code is in one package, so they can go straight from URL in the error response to the IResolvedUrl that hosts will need.

So I don't think we should codify the intermediate `redirectLocation` in the base driver code.